server/container/guice/mailet/src/test/java/org/apache/james/utils/GuiceMailetLoaderTest.java

```diff
@@ -183,4 +184,25 @@ public void getMailetShouldAllowCustomInjections() throws Exception {
 
         assertThatCode(() -> mailet.service(FakeMail.defaultFakeMail())).doesNotThrowAnyException();
     }
+
+    @Ignore("JAMES-2866 singleton are not shared between extensions")
+    @Test
+    public void allMailetsShouldShareTheSameSingleton() throws Exception {
```

Comment:
what about using ShrinkWrap to remove the pain of building the extension archive ( http://arquillian.org/arquillian-core/#deployment-archives ) ?